### PR TITLE
updated documentation for user.py and app.py 

### PR DESCRIPTION
**Pull request by wgranados**
also made exception hander in user.py a bit more precise

### CODE REVIEW

File: user.py [hunk 1]
```diff
@@ -1,24 +1,51 @@
...
+    """ Container class for basic user information collected from rooms.
+     
+    This information consists of user.id, user.rank, and user.name. user.id is
+    a format-removed id of user.name with only a-z lowercase and 0-9 present.
+
+    user.rank contain the auth level of the user, as a single character string of
+    either ' ', +, ☆, %, @, *, &, #, or ~. Note that ☆ is only relevant for 
+    battle rooms. 
+    
+    To compare groups against each other self.Groups have the information required
```
Comment (QuiteQuiet) [inline user.py]: I know it's in the old comment you removed, but `self.Groups` shouldn't be used, ever, over `User.Groups`

File: user.py [hunk 2]
```diff
@@ -1,24 +1,51 @@
...
+        except KeyError:
+            if rank1 not in User.Groups:
                 print('{rank} is not a supported usergroup'.format(rank = rank1))
-            if not rank2 in User.Groups:
+            if rank2 not in User.Groups:
                 print('{rank} is not a supported usergroup'.format(rank = rank2))
             return False
 
-    def __init__(self, name, rank = ' ', owner = False):
+    def __init__(self, name, rank=' ', owner=False):
```
Comment (QuiteQuiet) [inline user.py]: This change is contrary to every other `__init__` definition in the codebase :|

File: user.py [hunk 3]
```diff
@@ -1,24 +1,51 @@
...
         self.name = name
         self.id = re.sub(r'[^a-zA-z0-9]', '', name).lower()
         self.rank = rank
         self.owner = owner
 
     def hasRank(self, rank):
         return self.owner or User.compareRanks(self.rank, rank)
+
```
Comment (QuiteQuiet) [inline user.py]: Why?

File: user.py (hunk 2 above)
Comment (wgranados) [reply user.py]: uh, idk :|

File: user.py (hunk 3 above)
Comment (wgranados) [reply user.py]: Most classes have a space between methods, so I made this change.

Comment (QuiteQuiet) [reply user.py]: https://github.com/QuiteQuiet/PokemonShowdownBot/blob/master/room.py#L42-L54

I've generally put spaces between method unless they have a lot in common or affect the same system. It's only /mostly/ arbitrary.

Comment (wgranados) [reply user.py]: Ah, I'll keep that in mind next time.

**Pull request merged**